contracts/libraries/SafeMigration.sol

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.7.0 <0.9.0;
+
+import {SafeStorage} from "../libraries/SafeStorage.sol";
+import {ISafe} from "../interfaces/ISafe.sol";
+
+/**
+ * @title Migration Contract for Safe Upgrade
+ * @notice This is a generic contract that facilitates the migration of a Safe and SafeL2 contracts.
+ *         The supported target Safe version is immutable and set in the constructor during the deployment of the contract.
+ *         This contract also supports migration with fallback handler update.
+ * @author @safe-global/safe-protocol
+ * @dev IMPORTANT: The library is intended to be used with the Safe standard proxy that stores the singleton address
+ *      at the storage slot 0. Use at your own risk with custom proxy implementations. The contract will allow invocations
+ *      to the migration functions only via delegatecall.
+ */
+contract SafeMigration is SafeStorage {
+    /**
+     * @notice Address of this contract
+     */
+    address public immutable MIGRATION_SINGLETON;
+    /**
+     * @notice Address of the Safe Singleton implementation
+     */
+    address public immutable SAFE_SINGLETON;
+    /**
+     * @notice Address of the Safe Singleton (L2) implementation
+     */
+    address public immutable SAFE_L2_SINGLETON;
+    /**
+     * @notice Addresss of the fallbackhandler
+     */
+    address public immutable SAFE_FALLBACK_HANDLER;
+
+    /**
+     * @notice Event indicating a change of master copy address.
+     * @param singleton New master copy address
+     */
+    event ChangedMasterCopy(address singleton);
+
+    /**
+     * @notice Modifier to make a function callable via delegatecall only.
+     * If the function is called via a regular call, it will revert.
+     */
+    modifier onlyDelegateCall() {
+        require(address(this) != MIGRATION_SINGLETON, "Migration should only be called via delegatecall");
+        _;
+    }
+
+    /**
+     * @notice Constructor
+     * @param safeSingleton Address of the Safe Singleton implementation
+     * @param safeL2Singleton Address of the SafeL2 Singleton implementation
+     * @param fallbackHandler Address of the standard fallback handler implementation
```

Comment:
```suggestion
     * @param fallbackHandler Address of the fallback handler implementation
```
I'm not sure if we should say standard here because we have multiple implementations (token callback handler and compatibility fallback handler). I think we can even narrow down it further and say that we use the compatibility fallback handler here.

Reply:
> I'm not sure if we should say standard here because we have multiple implementations (token callback handler and compatibility fallback handler). 

Fine with this suggestion.

> I think we can even narrow down it further and say that we use the compatibility fallback handler here.

The migration contract can also be used with token callback handler or even extensible fallback handler. Migration contract deployment does not require that fallback handler has to be compatibility fallback handler.

Reply:
I suggested added the word "standard". The reason I suggested it is that we are only deploying a migration contract with the compatibility fallback handler, and "standard" felt like the correct word.

Fine with removing it if others don't agree.